sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -174,13 +174,13 @@ private[sql] class SharedState(
 object SharedState extends Logging {
   @volatile private var fsUrlStreamHandlerFactoryInitialized = false
 
-  private def setFsUrlStreamHandlerFactory(conf: SparkConf): Unit = {
+  private def setFsUrlStreamHandlerFactory(ctx: SparkContext): Unit = {
     if (!fsUrlStreamHandlerFactoryInitialized &&
-        conf.get(DEFAULT_URL_STREAM_HANDLER_FACTORY_ENABLED)) {
+        ctx.conf.get(DEFAULT_URL_STREAM_HANDLER_FACTORY_ENABLED)) {
       synchronized {
         if (!fsUrlStreamHandlerFactoryInitialized) {
           try {
-            URL.setURLStreamHandlerFactory(new FsUrlStreamHandlerFactory())
+            URL.setURLStreamHandlerFactory(new FsUrlStreamHandlerFactory(ctx.hadoopConfiguration))
```

Comment:
Can we narrow down to pass `hadoopConfiguration` instead of SparkContext? For the existing `conf: SparkConf`, you can keep it.

Reply:
done